runtime/src/lib.rs

```diff
@@ -262,6 +271,58 @@ impl frame_system::Config for Runtime {
 	type MaxConsumers = frame_support::traits::ConstU32<16>;
 }
 
+pub struct TagProvider;
+impl pallet_account_alias_registry::TagProvider<Runtime> for TagProvider {
+	fn tag(id: &AccountId, name: &str) -> Result<u16, ()> {
+		let salt = pallet_timestamp::Pallet::<Runtime>::get() / 86400000u64;
+		Self::tag_inner(id.as_slice(), name, salt)
+	}
+}
+impl TagProvider {
+	fn tag_inner(id: &[u8], name: &str, salt: u64) -> Result<u16, ()> {
+		let hash = (name, id, salt).using_encoded(sp_io::hashing::blake2_256);
```

Comment:
```suggestion
		let hash = (id, name, salt).using_encoded(sp_io::hashing::blake2_256);
```

blake2 is immune to length extension. You can encode raw data in the sequence of `(id, name, salt)`.